packages/pigeon/lib/objc_generator.dart

```diff
@@ -79,37 +79,45 @@ const Map<String, String> _objcTypeForDartTypeMap = <String, String>{
   'Map': 'NSDictionary',
 };
 
-const Map<String, String> _propertyTypeForDartTypeMap = <String, String>{
-  'String': 'copy',
-  'bool': 'strong',
-  'int': 'strong',
-  'double': 'strong',
-  'Uint8List': 'strong',
-  'Int32List': 'strong',
-  'Int64List': 'strong',
-  'Float64List': 'strong',
-  'List': 'strong',
-  'Map': 'strong',
-};
+String _flattenTypeArguments(String? classPrefix, List<TypeDeclaration> args) {
+  final String result = args
+      .map((TypeDeclaration e) => '${_objcTypeForDartType(classPrefix, e)} *')
+      .reduce((String value, String element) => '$value, $element');
+  return result;
+}
 
-String? _objcTypePtrForPrimitiveDartType(String type) {
-  return _objcTypeForDartTypeMap.containsKey(type)
-      ? '${_objcTypeForDartTypeMap[type]} *'
+String? _objcTypePtrForPrimitiveDartType(String? classPrefix, NamedType field) {
+  return _objcTypeForDartTypeMap.containsKey(field.dataType)
+      ? '${_objcTypeForDartType(classPrefix, field)} *'
       : null;
 }
 
-/// Returns the objc type for a dart [type], prepending the [classPrefix] for
-/// generated classes.  For example:
-/// _objcTypeForDartType(null, 'int') => 'NSNumber'.
-String _objcTypeForDartType(String? classPrefix, String type) {
-  final String? builtinObjcType = _objcTypeForDartTypeMap[type];
-  return builtinObjcType ?? _className(classPrefix, type);
+String _objcTypeForDartType<T extends TypedEntity>(
+    String? classPrefix, T field) {
+  return _objcTypeForDartTypeMap.containsKey(field.dataType)
+      ? field.typeArguments == null
+          ? _objcTypeForDartTypeMap[field.dataType]!
+          : '${_objcTypeForDartTypeMap[field.dataType]}<${_flattenTypeArguments(classPrefix, field.typeArguments!)}>'
+      : _className(classPrefix, field.dataType);
 }
 
-String _propertyTypeForDartType(String type) {
-  final String? result = _propertyTypeForDartTypeMap[type];
+String _propertyTypeForDartType(NamedType field) {
+  const Map<String, String> propertyTypeForDartTypeMap = <String, String>{
+    'String': 'copy',
+    'bool': 'strong',
+    'int': 'strong',
+    'double': 'strong',
+    'Uint8List': 'strong',
+    'Int32List': 'strong',
+    'Int64List': 'strong',
+    'Float64List': 'strong',
+    'List': 'strong',
+    'Map': 'strong',
+  };
+
+  final String? result = propertyTypeForDartTypeMap[field.dataType];
   if (result == null) {
-    return 'assign';
+    return 'strong';
```

Comment:
Why the behavioral change here?

Reply:
Just juggled the function.  The code above now catches the assign cases.  If we get there it should be a custom type.